### PR TITLE
Changed parameter default values

### PR DESCRIPTION
**Pull request by mcasperson**
Two previous steps needed to be updated with more sensible default values. The first change is to use the s3 terraform backend by default, and the second is to use the space id as the workspace name.

# Pre-requisites

- [ ] `Id` should be a **GUID** that is not `00000000-0000-0000-0000-000000000000`
  - **NOTE** If you are modifying an existing step template, please make sure that you **do not** modify the `Id` property *(updating the `Id` will break the Library sync functionality in Octopus)*. 
- [ ] `Version` should be incremented, otherwise the integration with Octopus won't update the step template correctly
- [ ] Parameter names should not start with `$`
- [ ] **Step template parameter names (the ones declared in the JSON, not the script body) should be prefixed with a namespace so that they are less likely to clash with other user-defined variables in Octopus** (see [this issue](https://github.com/OctopusDeploy/Issues/issues/2126)). For example, use an abbreviated name of the step template or the category of the step template).
- [ ] `LastModifiedBy` field must be present, and (_optionally_) updated with the correct author
- [ ] If a new `Category` has been created:
   - [ ] An image with the name `{categoryname}.png` must be present under the `step-templates/logos` folder
   - [ ] The `switch` in the `humanize` function in [`gulpfile.babel.js`](https://github.com/OctopusDeploy/Library/blob/master/gulpfile.babel.js#L92) must have a `case` statement corresponding to it

Fixes # . _If there is an open issue that this PR fixes add it here, otherwise just remove this line_

### CODE REVIEW

File: step-templates/octopus-apply-octoterra-module-s3.json [hunk 1]
```diff
@@ -126,7 +126,7 @@
       "Name": "OctoterraApply.AWS.S3.BucketKey",
       "Label": "AWS S3 Bucket Key",
       "HelpText": "The S3 file used to hold the Terraform state. See the [Terraform documentation](https://developer.hashicorp.com/terraform/language/settings/backends/s3) for details on using S3 as a backend. The combination of the workspace name and this key must be unique. \n\nThe default value is the name of the project: `#{Octopus.Project.Name | Replace \"[^A-Za-z0-9]\" \"_\"}`.",
-      "DefaultValue": "#{Octopus.Project.Name | Replace \"[^A-Za-z0-9]\" \"_\"}",
+      "DefaultValue": "Project_#{Octopus.Project.Name | Replace \"[^A-Za-z0-9]\" \"_\"}",
```
Comment (twerthi) [inline step-templates/octopus-apply-octoterra-module-s3.json:129]: Help Text seems to specify a different value than the DefaultValue attribute, DefaultValue is prefixing `Project_`

File: step-templates/octopus-apply-octoterra-module-s3.json [hunk 2]
```diff
@@ -45,8 +45,8 @@
       "Id": "27254625-8cfd-4918-b16b-68ac26a25d37",
       "Name": "OctoterraApply.Terraform.Workspace.Name",
       "Label": "Terraform Workspace",
-      "HelpText": "The name of the Terraform workspace. This must be unique for every project this module is deployed to. The default value is based on the space name: `#{Octopus.Space.Name | Replace \"[^A-Za-z0-9]\" \"_\"}`",
-      "DefaultValue": "#{Octopus.Space.Name | Replace \"[^A-Za-z0-9]\" \"_\"}",
+      "HelpText": "The name of the Terraform workspace. This must be unique for every project this module is deployed to. The default value is based on the space ID that the module is applied to: `#{OctoterraApply.Octopus.SpaceID}`. Leave this as the default value unless you have a specific reason to change it.",
```
Comment (twerthi) [inline step-templates/octopus-apply-octoterra-module-s3.json:48]: The value of this parameter is defaulted to another parameter that doesn't have a default value, I'm assuming this is intentional?

Comment (mcasperson) [reply step-templates/octopus-apply-octoterra-module-s3.json:48]: Yes, 99% of the time this value is just the space ID. There will be edge cases where this value is not unique enough, usually when applying values across multiple instances that may have duplicate space IDs, which is why it is a configurable field. But the default is to reuse an internal parameter.

File: step-templates/octopus-apply-octoterra-module-s3.json (hunk 1 above)
Comment (mcasperson) [reply step-templates/octopus-apply-octoterra-module-s3.json:129]: :+1: I'll update that text.

**Review by twerthi — APPROVED** [review]
LGTM

**Pull request merged**